src/plugins/intel_gpu/src/kernel_selector/cl_kernels/softmax_gpu_bf.cl

```diff
@@ -35,6 +35,30 @@
 
 #endif
 
+inline uint FUNC(index_calc)(uint idx, bool use_output_buffer, uint aligned_data_offset) {
+#if IS_DYNAMIC
+    if (use_output_buffer) {
+        return (aligned_data_offset + get_sub_group_local_id() + idx * get_sub_group_size());
+    } else {
+        return idx;
+    }
```

Comment:
please extract out static values calculation aligned_data_offset + get_sub_group_local_id() from the loop.
e.g., 
int data_offset = (use_output_buffer) ? aligned_data_offset + get_sub_group_local_id() : 0;
int idx_stride = (use_output_buffer) ? get_sub_group_size() : 1;

unroll_for(...) {
   tmp_ptr[data_offset + idx_stride * idx] = ...
}

Reply:
@yeonbok Updated index calculation as your suggestion. Thanks!